test-app/tests/integration/components/input-test.gts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-undef -- Until https://github.com/ember-cli/eslint-plugin-ember/issues/1747 is resolved... */
+/* eslint-disable simple-import-sort/imports,padding-line-between-statements,decorator-position/decorator-position -- Can't fix these manually, without --fix working in .gts */
+
+import { fillIn, render } from '@ember/test-helpers';
+import { module, test } from 'qunit';
+
+// outside of tests, in everyday use, use <Form::Controls::Input />
+import InputControl from '@crowdstrike/ember-toucan-core/components/form/controls/input';
+import { setupRenderingTest } from 'test-app/tests/helpers';
+
+module('Integration | Component | Input', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(<template>
+      {{! we do not require a label, but instead suggest using Field / TextareaField }}
+      {{! template-lint-disable require-input-label }}
+      <InputControl data-input />
+    </template>);
+
+    assert.dom('[data-input]').hasTagName('input');
+    assert.dom('[data-input]').hasClass('text-titles-and-attributes');
+    assert.dom('[data-input]').doesNotHaveClass('text-disabled');
+  });
+
+  test('it disables the textarea using `@isDisabled`', async function (assert) {
+    await render(<template>
+      {{! we do not require a label, but instead suggest using Field / TextareaField }}
+      {{! template-lint-disable require-input-label }}
+      <InputControl @isDisabled={{true}} data-input />
+    </template>);
+
+    assert.dom('[data-input]').isDisabled();
+    assert.dom('[data-input]').hasClass('text-disabled');
+    assert
+      .dom('[data-input]')
+      .doesNotHaveClass('text-titles-and-attributes');
+  });
+
+  test('it spreads attributes to the underlying textarea', async function (assert) {
+    await render(<template>
+      {{! we do not require a label, but instead suggest using Field / TextareaField }}
+      {{! template-lint-disable require-input-label }}
+      <InputControl placeholder="Placeholder text" data-input />
```

Comment:
Interesting, how does template-lint know this is representing an `<input>`? By the component name? The component could render something totally different, where that rule does not apply... 🤔

Reply:
Oh interesting! I had to do the same thing with Textarea - may you're right that it is by the component name. I'd be curious!  I'd also be impressed if it dug into the template to find it was in fact an `<input>`!

Reply:
>Interesting, how does template-lint know this is representing an <input>? By the component name? The component could render something totally different, where that rule does not apply... 🤔

I think it visits all ElementNodes and looks for any instance of say, `Input` or `input`...if it sees `input` it then checks the parent's attributes.
<img width="1227" alt="Screen Shot 2023-02-27 at 1 25 19 PM" src="https://user-images.githubusercontent.com/771170/221651047-b4ec44c5-ba8b-4dd5-819c-e1156c4359f2.png">


https://github.com/ember-template-lint/ember-template-lint/blob/master/lib/rules/require-input-label.js#L88

Reply:
Yeah, but our tag here is `InputControl`, which is not part of [this SET](https://github.com/ember-template-lint/ember-template-lint/blob/master/lib/rules/require-input-label.js#L29). So still seems weird to me. 🤔 

Anyway, nothing that should block us here, just curious...

Reply:
I'm not totally understanding what's going on in the lint rule code, but [in the tests](https://github.com/ember-template-lint/ember-template-lint/blob/master/test/unit/rules/require-input-label-test.js#L74) it looks like it searches for any `input` and checks the surrounding (parent?) tags to see if there is an `label`. With `InputControl` it can tell there is an `input` inside but no label.